plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py

```diff
@@ -90,3 +92,49 @@ def string_to_unicode(data):
         except UnicodeError:
             # try others
             continue
+
+
+def package_headers(filename):
+    """
+    Return package header.
+
+    :param filename: full path to the package to analyze
+    :type  filename: str
+
+    :return: package header
+    :rtype: rpm.hdr
+    """
+
+    # Read the RPM header attributes for use later
+    ts = rpm_module.TransactionSet()
+    ts.setVSFlags(rpm_module._RPMVSF_NOSIGNATURES)
+    fd = os.open(filename, os.O_RDONLY)
```

Comment:
I assume the below try/except is there because ``os.open`` doesn't return a context manager. You could instead do this:

```
with contextlib.closing(os.open(filename, os.O_RDONLY)) as fd:
  headers = ts.hdrFromFdno(fd)
```